api_tests/preprint_providers/views/test_preprint_provider_moderator_list.py

```diff
@@ -0,0 +1,113 @@
+import pytest
+
+from api.base.settings.defaults import API_BASE
+from api.preprint_providers.permissions import GroupHelper
+from osf_tests.factories import (
+    AuthUserFactory,
+    PreprintProviderFactory,
+)
+
+
+@pytest.mark.django_db
+class TestPreprintProviderModeratorList:
+
+    @pytest.fixture()
+    def provider(self):
+        pp = PreprintProviderFactory(name='ModArxiv')
+        GroupHelper(pp).update_provider_auth_groups()
+        return pp
+
+    @pytest.fixture()
+    def url(self, provider):
+        return '/{}preprint_providers/{}/moderators/'.format(API_BASE, provider._id)
+
+    @pytest.fixture()
+    def admin(self, provider):
+        user = AuthUserFactory()
+        GroupHelper(provider).get_group('admin').user_set.add(user)
+        return user
+
+    @pytest.fixture()
+    def moderator(self, provider):
+        user = AuthUserFactory()
+        GroupHelper(provider).get_group('moderator').user_set.add(user)
+        return user
+
+    @pytest.fixture()
+    def nonmoderator(self):
+        return AuthUserFactory()
+
+    def create_payload(self, user_id, permission_group, full_name=None):
+        data = {
+            'data': {
+                'attributes': {
+                    'permission_group': permission_group,
+                },
+                'type': 'moderators',
+                'id': user_id
+            }
+        }
+        if full_name:
+            data['data']['attributes']['full_name'] = full_name
+        return data
+
+    def test_list_get_not_authorized(self, app, url, nonmoderator, admin, provider):
+        res = app.get(url, expect_errors=True)
+        assert res.status_code == 401
+
+        res = app.get(url, auth=nonmoderator.auth, expect_errors=True)
+        assert res.status_code == 403
+
+    def test_list_get_moderator(self, app, url, nonmoderator, moderator, admin, provider):
+        # admin/nonmoderator unused here, just len verification
+        res = app.get(url, auth=moderator.auth)
+        assert res.status_code == 200
+        assert len(res.json['data']) == 2
+        for datum in res.json['data']:
+            if datum['id'] == moderator._id:
+                assert datum['attributes']['permission_group'] == 'moderator'
+
+    def test_list_get_admin_with_filter(self, app, url, nonmoderator, moderator, admin, provider):
```

Comment:
could you add one for id filter... I feel like that fails the most.

Reply:
I can add `id` to this filter. I tried to write these in a way that wouldn't incur a lot of set-up / tear-down time

Reply:
Yeah they look great in that regard-- doing the double filter is fine by me (and something we do not test frequently iirc)